src/openrct2/peep/Staff.cpp

```diff
@@ -84,16 +84,24 @@ static inline void staff_autoposition_new_staff_member(rct_peep * newPeep)
 
     newPeep->state = PEEP_STATE_FALLING;
 
-    sint16     x, y, z;
-    uint32     count = 0;
-    uint16     sprite_index;
-    rct_peep * guest = nullptr;
+    sint16              x, y, z;
+    uint32              count = 0;
+    uint16              sprite_index;
+    sint32              tile_type;
+    rct_peep *          guest = nullptr;
+    rct_tile_element *  guest_tile = nullptr;
 
     // Count number of walking guests
     FOR_ALL_GUESTS(sprite_index, guest)
     {
         if (guest->state == PEEP_STATE_WALKING)
-            ++count;
+        {
+            // check the walking guest's tile, only count them if they're on a path tile
+            guest_tile = map_get_path_element_at(guest->x / 32, guest->y / 32, guest->next_z);
+            tile_type = (guest_tile != nullptr) ? tile_element_get_type(guest_tile) : 0;
```

Comment:
There is no need to check the type that's what get path does. It's either nullptr or its a path tile. I'd also hesitate to use x,y, next_z. Stick to all next or all x,y,z. But then I'm going to refactor this to next flags so it's not really that important.

Reply:
Yeah, that was a leftover check for when I was trying a different function for getting the tile. I can refactor to the next flags, for some reason the z flag returned no tile at all from the get path function.

Reply:
Doesn't `guest->next_z` need to be divided by 8 here, to map the big coords to small coords?

Reply:
I see. I saw that we divided the x,y,z coordinates for mapping, but I wasn't sure if the next ones need to be converted. The testing I did succeeded, though I may have been too narrow in my placements.